modules/apps/portal/portal-http-impl/src/test/java/com/liferay/portal/http/internal/HttpImplTest.java

```diff
@@ -71,6 +151,127 @@ public void testIsNonProxyHost() throws Exception {
 		}
 	}
 
+	private Tuple _getHttpImplConnectionStrategies() {
+		CloseableHttpClient closeableHttpClient = ReflectionTestUtil.invoke(
+			_httpImpl, "getCloseableHttpClient",
+			new Class<?>[] {HttpHost.class}, new Object[] {null});
+
+		ClientExecChain clientExecChain = ReflectionTestUtil.getFieldValue(
+			closeableHttpClient, "execChain");
+
+		while (true) {
+			clientExecChain = ReflectionTestUtil.getFieldValue(
+				clientExecChain, "requestExecutor");
+
+			if (clientExecChain instanceof MainClientExec) {
+				ConnectionKeepAliveStrategy connectionKeepAliveStrategy =
+					ReflectionTestUtil.getFieldValue(
+						clientExecChain, "keepAliveStrategy");
+				ConnectionReuseStrategy connectionReuseStrategy =
+					ReflectionTestUtil.getFieldValue(
+						clientExecChain, "reuseStrategy");
+
+				return new Tuple(
+					connectionKeepAliveStrategy, connectionReuseStrategy);
+			}
+		}
+	}
+
+	private void _resetHttpImpl() {
+		DCLSingleton<CloseableHttpClient> closeableHttpClientDCLSingleton =
+			ReflectionTestUtil.getFieldValue(
+				_httpImpl, "_closeableHttpClientDCLSingleton");
+
+		ReflectionTestUtil.setFieldValue(
+			closeableHttpClientDCLSingleton, "_singleton", null);
+
+		DCLSingleton<PoolingHttpClientConnectionManager>
+			poolingHttpClientConnectionManagerDCLSingleton =
+				ReflectionTestUtil.getFieldValue(
+					_httpImpl,
+					"_poolingHttpClientConnectionManagerDCLSingleton");
+
+		ReflectionTestUtil.setFieldValue(
+			poolingHttpClientConnectionManagerDCLSingleton, "_singleton", null);
+	}
+
+	private void _testHttpKeepAlive(
+		boolean expectedKeepAlive, long expectedKeepAliveTimeoutInMilliseconds,
+		HttpContext httpContext, HttpResponse httpResponse) {
+
+		try {
+			Tuple connectionStrategiesTuple =
+				_getHttpImplConnectionStrategies();
+
+			ConnectionKeepAliveStrategy connectionKeepAliveStrategy =
+				(ConnectionKeepAliveStrategy)
+					connectionStrategiesTuple.getObject(0);
+			ConnectionReuseStrategy connectionReuseStrategy =
+				(ConnectionReuseStrategy)connectionStrategiesTuple.getObject(1);
+
+			Assert.assertNotNull(connectionKeepAliveStrategy);
+			Assert.assertNotNull(connectionReuseStrategy);
+
+			Assert.assertEquals(
```

Comment:
@ericyanLr, please remove the below assert because they are not our test target and the below logic will throw NPE if they are null.
```
Assert.assertNotNull(connectionKeepAliveStrategy);
Assert.assertNotNull(connectionReuseStrategy);
```